pkg/compose/build_buildkit.go

```diff
@@ -71,3 +82,197 @@ func (s *composeService) doBuildBuildkit(ctx context.Context, project *types.Pro
 
 	return imagesBuilt, err
 }
+
+func (s *composeService) getDrivers(ctx context.Context) ([]build.DriverInfo, error) { //nolint:gocyclo
+	txn, release, err := storeutil.GetStore(s.dockerCli)
+	if err != nil {
+		return nil, err
+	}
+	defer release()
+
+	ng, err := storeutil.GetCurrentInstance(txn, s.dockerCli)
+	if err != nil {
+		return nil, err
+	}
+
+	dis := make([]build.DriverInfo, len(ng.Nodes))
+	var f driver.Factory
+	if ng.Driver != "" {
+		factories := driver.GetFactories()
+		for _, fac := range factories {
+			if fac.Name() == ng.Driver {
+				f = fac
+				continue
+			}
+		}
+		f = driver.GetFactory(ng.Driver, true)
```

Comment:
Looks like this will overwrite any factory set in the loop, do you want an `if f == nil` guard?

Reply:
yes we can